cdap-docs/build.sh

```diff
@@ -38,227 +66,197 @@ function set_project_path() {
   echo ""
-  echo "========================================================"
-  echo "Building GitHub Docs."
+  build_javadocs_api
   echo "--------------------------------------------------------"
-  echo ""
-  run_command docs-github-part ${ARG_2} ${ARG_3}
-  echo "Stashing GitHub Docs."
-  cd ${SCRIPT_PATH}
-  mkdir -p ${SCRIPT_PATH}/${BUILD_TEMP}
-  mv ${SCRIPT_PATH}/${BUILD}/*.zip ${SCRIPT_PATH}/${BUILD_TEMP}
-  echo ""
+  echo "Completed Build of Javadocs"
   echo "========================================================"
-  echo "Building Web Docs."
-  echo "--------------------------------------------------------"
-  echo ""
-  run_command docs-web-part ${ARG_2} ${ARG_3}
   echo ""
+}
+
+function build_javadocs_api() {
+  set_mvn_environment
+  MAVEN_OPTS="-Xmx1024m -XX:MaxPermSize=128m" mvn clean install -P examples,templates,release -DskipTests -Dgpg.skip=true && mvn clean site -DskipTests -P templates -DisOffline=false
+}
+
+function build_docs_first_pass() {
   echo "========================================================"
-  echo "Replacing GitHub Docs."
+  echo "Building First Pass of Docs"
   echo "--------------------------------------------------------"
   echo ""
-  mv ${SCRIPT_PATH}/${BUILD_TEMP}/*.zip ${SCRIPT_PATH}/${BUILD}
-  rm -rf ${SCRIPT_PATH}/${BUILD_TEMP}
-  echo ""
+  build_docs_inner_level "docs"
   echo "--------------------------------------------------------"
-  bell "Completed \"build_all\"."
-  echo "========================================================"
+  echo "Completed Building First Pass of Docs"
   echo "========================================================"
   echo ""
-  exit 0
-}
-
-function build_javadocs() {
-  # Uses function from common
-  build_javadocs_api
+  return $?
 }
 
-function build_docs_javadocs() {
-  build_docs_inner_level "build"
+function build_docs_github_part() {
+  echo "========================================================"
+  echo "Building GitHub Docs"
+  echo "--------------------------------------------------------"
+  echo ""
+  _build_docs "build-github" ${GOOGLE_ANALYTICS_GITHUB} ${GITHUB} ${FALSE}
+  return $?
 }
 
-function build_docs() {
-  _build_docs "docs" ${GOOGLE_ANALYTICS_WEB} ${WEB} ${TRUE}
-  return $?
+function _stash_github_zip() {
+  echo "========================================================"
+  echo "Stashing GitHub Zip"
+  echo "========================================================"
+  echo ""
+  cd ${SCRIPT_PATH}
+  TARGET_TEMP=$(mktemp -d /tmp/cdap-docs.XXXX)
+  mv ${SCRIPT_PATH}/${TARGET}/*.zip ${TARGET_TEMP}
+  echo ""
 }
 
-function build_docs_github() {
-  _build_docs "build-github" ${GOOGLE_ANALYTICS_GITHUB} ${GITHUB} ${FALSE}
-  return $?
+function _restore_github_zip() {
+  echo "========================================================"
+  echo "Restoring GitHub Zip"
+  echo "========================================================"
+  echo ""
+  mv ${TARGET_TEMP}/*.zip ${SCRIPT_PATH}/${TARGET}
+  rm -rf ${TARGET_TEMP}
```

Comment:
Rather than doing cleanup inside functions where a previous command could fail, it would make sense to instead trap signals and have an exit handler which performs cleanup.

```
trap trap_cleanup SIGHUP SIGQUIT SIGINT SIGTERM SIGKILL
function trap_cleanup() { cleanup; exit 1; }
function cleanup() {
  # perform cleanups here
}
```